src/UhdmAst.h

```diff
@@ -18,6 +18,7 @@ struct UhdmShared {
     std::string package_prefix;
     NameNodeMap partial_modules;
     std::unordered_map<const UHDM::BaseClass*, std::string> visited_types;
+    std::unordered_map<const UHDM::BaseClass*, std::string> visited_variables;
```

Comment:
As you mentioned, `unordered_set` would be fine here. Maybe rename to `visited_objects`?

Reply:
I think that the name should be changed, but I don't know if `visited_objects` would be good. We still need `visited_types` map and types are kind of objects. I don't have any idea for proper name.

Reply:
Maybe rename the `visited_types` to `visited_types_map` (like `top_param_map`)?

Actually we might reuse this set later if we revisit coverage tracking. I'm not dead set on making this `visited_objects`, but `visited_variables` is unacceptable for storing function object pointers.

Reply:
Ok, I will do it. Thank you.